core/sql/generator/GenExplain.cpp

```diff
@@ -660,6 +660,269 @@ FileScan::addSpecificExplainInfo(ExplainTupleMaster *explainTuple,
   return(explainTuple);
 }
 
+static void appendListOfColumns(Queue* listOfColNames,ComTdb *tdb, NAString& outNAString){
+
+	if (((ComTdbHbaseAccess*)tdb)->sqHbaseTable()){// if trafodion table
+	  char buf[1000];
+
+	  listOfColNames->position();
+	  for (Lng32 j = 0; j < listOfColNames->numEntries(); j++)
+	    {
+	      char * currPtr = (char*)listOfColNames->getCurr();
+
+	      Lng32 currPos = 0;
+	      Lng32 jj = 0;
+	      short colNameLen = *(short*)currPtr;
+	      currPos += sizeof(short);
+	      char colFam[100];
```

Comment:
I got that limit and code snippet  from the file ComTdbHbaseAccess.cpp: static void showStrColNames(Queue * listOfColNames, Space * space,
			    NABoolean nullTerminated = FALSE)
{
  char buf[1000];

  listOfColNames->position();
  for (Lng32 j = 0; j < listOfColNames->numEntries(); j++)
    {
      char * currPtr = (char*)listOfColNames->getCurr();
      
      char * colNamePtr = NULL;
      if (nullTerminated)
	{
	  colNamePtr = currPtr;
	}
      else
	{
	  Lng32 currPos = 0;
	  short colNameLen = *(short*)currPtr;
	  currPos += sizeof(short);
	  char colName[500];
	  
	  for (Lng32 i = 0; i < colNameLen; i++)
	    {
	      colName[i] = currPtr[currPos];
	      currPos++;
	    }
	  
	  colName[colNameLen] = 0;
	  
	  colNamePtr = colName;
	}

      str_sprintf(buf, "  Entry #%d: %s",
		  j+1, 
		  colNamePtr);
      
      space->allocateAndCopyToAlignedSpace(buf, str_len(buf), sizeof(short));

      listOfColNames->advance();
    } // for
}

So 3 options: 1/ I fix it on my code and we have code inconsistency. 2/ I leave is as is and open a JIRA so that is get fixed in both place: That way we have no inconsistency 3/ someone knows that original code was fine and limit is reasonable? and we don t touch any of it?